events/renderers.py

```diff
@@ -12,3 +26,198 @@ class JSONLDRenderer(JSONRenderer):
     media_type = 'application/ld+json'
     format = 'json-ld'
     charset = 'utf-8'
+
+
+class DOCXRenderer(renderers.BaseRenderer):
+    media_type = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'
+    format = 'docx'
+    charset = None
+    render_style = 'binary'
+
+    def get_query_params(self, renderer_context, param_name):
+        return renderer_context['request'].query_params.get(param_name, '')
+
+    def get_any_language(self, dictionary, default='fi', language_codes=None):
+        if not language_codes:
+            # Default order for when a language isn't found.
+            language_codes = list(set([default, 'fi', 'sv', 'en']))
+
+        for language_code in language_codes:
+            content = dictionary.get(language_code)
+            if content:
+                return content
+        return ''
+
+    def get_date_range(self, renderer_context):
+        dates = {'start': None, 'end': None}
+
+        start_date = self.get_query_params(renderer_context, 'start')
+        if start_date:
+            dates['start'] = dateutil.parser.parse(start_date)
+
+        end_date = self.get_query_params(renderer_context, 'end')
+        if end_date:
+            dates['end'] = dateutil.parser.parse(end_date)
+
+        try:
+            days = int(self.get_query_params(renderer_context, 'days'))
+        except ValueError as e:
+            days = None
+        # Days override dates
+        if days:
+            dates['start'] = datetime.datetime.now().date()
+            dates['end'] = dates['start'] + datetime.timedelta(days=days)
+
+        return dates
+
+    def render(self, data, media_type=None, renderer_context=None):
+        language_code = getattr(settings, 'LANGUAGE_CODE', 'fi')
+        document = Document()
+
+        location = self.get_query_params(renderer_context, 'location')
+        # This could be set to something like "All locations"
+        place_name = ''
+        try:
+            place_name = Place.objects.values_list(
+                'name',
+                flat=True,
+            ).get(id=location)
+        except Place.DoesNotExist:
+            raise Http404('Location not found')
```

Comment:
Currently, this exception is raised by a simple query `/v1/event/?format=docx`

It would be preferable that *any* event endpoint query may be rendered in the docx format. This makes the api more consistent. Therefore, if location, dates etc. are not specified, the docx should just leave that metadata empty.